front/src/App.tsx

```diff
@@ -12,10 +12,37 @@ const INITIAL_CHARACTER: Character = {
     class: ClassName.Sorcerer,
     spellCastingModifier: 0,
     showSpellSaveDC: true,
+    id: "",
 };
 
+const CHARACTERS_KEY = "characters";
+
+function getLocallyStoredCharacter(): Character {
+    const locallyStoredCharactersJSON = localStorage.getItem(CHARACTERS_KEY);
+    if (locallyStoredCharactersJSON === null) {
+        return { ...INITIAL_CHARACTER, id: crypto.randomUUID() };
+    }
+    const locallyStoredCharacters: Character[] = JSON.parse(
+        locallyStoredCharactersJSON
+    );
+    const firstCharacter = locallyStoredCharacters[0];
+    return {
+        name: firstCharacter.name,
+        class: firstCharacter.class,
+        spellCastingModifier: firstCharacter.spellCastingModifier,
+        showSpellSaveDC: firstCharacter.showSpellSaveDC,
+        id: firstCharacter.id,
+    };
```

Comment:
```suggestion
    const locallyStoredCharacters = JSON.parse(localStorage.getItem(CHARACTERS_KEY));
    
    if (!locallyStoredCharacters) {
        return { ...INITIAL_CHARACTER, id: crypto.randomUUID() };
    }

    return locallyStoredCharacters[0];
```

We can just check `!locallyStoredCharacters` to rule out if it's `null` or `[]`.

Reply:
nice nice